packages/ens-controller/src/EnsController.ts

```diff
@@ -86,6 +118,25 @@ export class EnsController extends BaseControllerV2<
         ...state,
       },
     });
+
+    if (provider && onNetworkStateChange) {
+      onNetworkStateChange((networkState) => {
```

Comment:
We need to clear the `ensResolutionsByAddress` state upon each network switch, because the values are not valid across networks. This is how the extension controller works today.

Reply:
It looks like this is still missing; there should be a `resetState()` call here

Reply:
Thanks, this seems to be resolved now!

Nit: It would be great to have this behavior covered by a test. We could have a test that shows that after the network is switched, the state gets cleared.